cli/setup.sh

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env bash
+
+[[ -n "${GENERATION_DEBUG}" ]] && set ${GENERATION_DEBUG}
+trap 'exit ${RESULT:-1}' EXIT SIGHUP SIGINT SIGTERM
+. "${GENERATION_BASE_DIR}/execution/common.sh"
+
+
+function update_provider_state() {
+    local state="$1"; shift
+    local provider_id="$1"; shift
+    local provider_type="$1"; shift
+    local provider_ref="$1"; shift
+    local plugin_dir="$1"; shift
+
+    provider_ref="${provider_type}:${provider_ref}"
+    echo "${state}" | jq -r --arg id "${provider_id}" --arg type "${provider_type}" --arg ref "${provider_ref}" --arg plugin_dir "${plugin_dir}" \
+                          '.Providers[$id] = { type: $type, ref: $ref, plugin_dir: $plugin_dir }'
+}
+
+function options() {
+
+  # Parse options
+  while getopts ":hi:" option; do
+      case "${option}" in
+          i|p) TEMPLATE_ARGS="${TEMPLATE_ARGS} -${option} ${OPTARG}" ;;
```

Comment:
Is this missing the getopts input for `p` ?

Reply:
Fixed